keypoints/src/brisk_2d.cpp

```diff
@@ -1669,7 +1669,7 @@ pcl::keypoints::brisk::Layer::halfsample (
       // compute horizontal pairwise average and store
       p_dest_char = reinterpret_cast<unsigned char*> (p_dest);
 #ifdef __GNUC__
-      typedef unsigned char __attribute__ ((__may_alias__)) UCHAR_ALIAS;
+      using UCHAR_ALIAS = unsigned char;
```

Comment:
Any idea what did this attribute do?

Reply:
I have the feeling it likely disables aliasing warnings that will be triggered by this. https://github.com/PointCloudLibrary/pcl/blob/2f0dc496dfb53dc68dde72b7e631aa1cf7bb833b/keypoints/src/brisk_2d.cpp#L1678

More on the topic:
https://stackoverflow.com/questions/6313050/gcc-how-to-use-attribute-may-alias-properly-to-avoid-derefencing-type
https://en.wikipedia.org/wiki/Aliasing_(computing)

Reply:
Let's maybe leave the typedef here?

Reply:
I'm surprised the same logic can't be expressed with `using`.

Reply:
Maybe it could. But frankly speaking... I have no desire to study this question. Getting rid of each and every `typedef` is a no-goal for us, so leaving as-is is the safest option I think.

Reply:
@SunBlack according to [this page](https://en.cppreference.com/w/cpp/language/type_alias), `using` alias should look like:

```
using identifier attr(optional) = type-id 
```

That is, attributes come before equals sign. Right now attributes are before equals and the thing still compiles, however we don't know whether it works as expected. I'd suggest we keep `typedef` to avoid potential troubles.

Reply:
You are right, my code is not correct. I played now a bit with it by using a aliasing example from [here](https://gist.github.com/shafik/848ae25ee209f698763cffee272a58f8). I adjusted the code the to use aliasing typedef, see [here](https://godbolt.org/z/MPGMNI).

Result:
`typedef float FLOAT_ALIAS;` and `using FLOAT_ALIAS = __attribute__ ((__may_alias__)) float;` are producing:

```assembly
foo(float*, int*): # @foo(float*, int*)
  mov dword ptr [rsi], 1
  mov dword ptr [rdi], 0
  mov eax, 1
  ret
```

`typedef float __attribute__ ((__may_alias__)) FLOAT_ALIAS;` and `using FLOAT_ALIAS __attribute__ ((__may_alias__)) = float;` are producing:
```assembly
foo(float*, int*): # @foo(float*, int*)
  mov dword ptr [rsi], 1
  mov dword ptr [rdi], 0
  mov eax, dword ptr [rsi]
  ret
```

@taketwo  I believed `attr` in `using identifier attr(optional) = type-id` is only for new attributes like `[[deprecated]]`, but are right :).

Reply:
Just to add a note: I don't believe we need aliasing here anywhere, because we are just reading value of `result1` (so nothing to optimize there).

Reply:
:+1: thanks for researching assembly, good to know!